ui/conductor/src/routes/ops/notifications/useAlertsApi.js

```diff
@@ -204,6 +206,87 @@ export default function(name, query) {
     return fetchingPages.value !== false || pullResource.loading;
   });
 
+  // --------- Export data as CSV --------- //
+  const notifications = useNotifications();
+  const downloadListPageTracker = reactive(
+      /** @type {ActionTracker<ListAlertsResponse.AsObject>} */
+      newActionTracker()
+  );
+  const downloadNextPageToken = ref('');
+  let allItemsToDownload = [];
+
+  /**
+   * Exports the data as a CSV file.
+   * We process the existing records and convert them into a CSV format.
+   *
+   * @param {string} fileName
+   * @return {Promise<void>}
+   */
+  const exportData = async (fileName) => {
+    let notificationData = []; // data to be downloaded as CSV
+
+    // fetch the data
+    try {
+      await fetchMore(listQuery.value, queryVersionCounter.value);
+      downloadNextPageToken.value = nextPageToken.value;
+      allItemsToDownload.push(...allItems.value);
+
+      // if there is more data to fetch, we fetch it and add it to the existing records
+      while (downloadNextPageToken.value) {
+        const page = await listAlerts({
+          ...listQuery.value,
+          pageSize: 1000,
+          pageToken: downloadNextPageToken.value
+        }, downloadListPageTracker);
+
+        downloadNextPageToken.value = page.nextPageToken;
+        allItemsToDownload.push(...page.alertsList.map((a) => transform(a)));
+      }
+    } finally {
+      notificationData = allItemsToDownload.map((item) => {
+        // Initialize ackTime only if it exists and is valid
+        let ackTime = null;
+        if (item.acknowledgement && item.acknowledgement.acknowledgeTime) {
+          ackTime = item.acknowledgement.acknowledgeTime;
+        }
+
+        // Safely handle createTime and resolveTime by checking if they are defined
+        const createTimeString = item.createTime ?
+            `${item.createTime.toLocaleDateString()} ${item.createTime.toLocaleTimeString()}` :
+            '';
+        const resolveTimeString = item.resolveTime ?
+            `${item.resolveTime.toLocaleDateString()} ${item.resolveTime.toLocaleTimeString()}` :
+            '';
+        const ackTimeString = ackTime ? `${ackTime.toLocaleDateString()} ${ackTime.toLocaleTimeString()}` : '';
+
+        // Return the data to be downloaded as CSV in the correct format
+        return {
+          'Create Time': createTimeString,
+          'Source': item.source,
+          'Floor': item.floor,
+          'Zone': item.zone,
+          'Severity': notifications.severityData(item.severity).text,
+          'Description': item.description,
+          'Resolve Time': resolveTimeString,
+          'Acknowledged': item.acknowledgement ? 'Yes' : 'No',
+          'Acknowledged Time': ackTimeString
```

Comment:
Is there a reason you aren't including information about who acknowledged the alert?

Reply:
At the moment, I don't think we export acknowledged alerts, but I'll have a look!